cumulus/primitives/core/src/lib.rs

```diff
@@ -155,11 +155,19 @@ pub trait UpwardMessageSender {
 	/// be dispatched or an error if the message cannot be sent.
 	/// return the hash of the message sent
 	fn send_upward_message(msg: UpwardMessage) -> Result<(u32, XcmHash), MessageSendError>;
+
+	/// Check whether the message size is acceptable for the channel.
+	fn can_send_upward_message(size: usize) -> Result<(), MessageSendError>;
 }
+
 impl UpwardMessageSender for () {
 	fn send_upward_message(_msg: UpwardMessage) -> Result<(u32, XcmHash), MessageSendError> {
 		Err(MessageSendError::NoChannel)
 	}
+
+	fn can_send_upward_message(_size: usize) -> Result<(), MessageSendError> {
```

Comment:
What about changing `usize` to `&UpwardMessage`?
```suggestion
	fn can_send_upward_message(_msg: &UpwardMessage) -> Result<(), MessageSendError> {
```

Reply:
you are right, it makes a lot more sense like this!

Reply:
https://github.com/paritytech/polkadot-sdk/pull/8409/commits/6c39e811ac58f3c5ad46c3b07630ae43259002b0